### PR TITLE
feat(material/paginator): Switch paginator implementation to use MDC

### PR DESCRIPTION
**Pull request by andrewseguin**

### CODE REVIEW

**Comment by mmalerba** [comment]
looks like you need a rebase

**Comment by andrewseguin** [comment]
All set and ready for review

File: src/components-examples/material/paginator/BUILD.bazel [hunk 1]
```diff
@@ -42,8 +42,8 @@ ng_test_library(
         ":paginator",
         "//src/cdk/testing",
         "//src/cdk/testing/testbed",
-        "//src/material/paginator",
-        "//src/material/paginator/testing",
+        "//src/material/legacy-paginator",
```
Comment (mmalerba) [inline src/components-examples/material/paginator/BUILD.bazel:45]: need to invert the dep

Comment (andrewseguin) [reply src/components-examples/material/paginator/BUILD.bazel:45]: I thought with examples we were leaving them as-is?

Comment (mmalerba) [reply src/components-examples/material/paginator/BUILD.bazel:45]: Ah yep, too many build files in this PR and I got confused which one I was looking at. You're good

**Review by mmalerba — APPROVED** [review]
LGTM

**Review by wagnermaciel — APPROVED** [review]
lgtm

**Pull request merged**